go/vt/vtctl/workflow/utils.go

```diff
@@ -1011,3 +1011,56 @@ func applyTargetShards(ts *trafficSwitcher, targetShards []string) error {
 	}
 	return nil
 }
+
+// validateEmptyTables checks if all specified tables in the keyspace are empty across all shards.
+// It queries each shard's primary tablet and if any non-empty table is found, it returns an error
+// containing a list of non-empty tables.
+func validateEmptyTables(ctx context.Context, ts *topo.Server, tmc tmclient.TabletManagerClient, keyspace string, tables []string) error {
+	shards, err := ts.GetServingShards(ctx, keyspace)
+	if err != nil {
+		return err
+	}
+	if len(shards) == 0 {
+		return fmt.Errorf("keyspace %s has no shards", keyspace)
+	}
+
+	isFaultyTable := map[string]bool{}
+	for _, shard := range shards {
+		primary := shard.PrimaryAlias
+		if primary == nil {
+			return fmt.Errorf("shard does not have a primary: %v", shard.ShardName())
+		}
+
+		ti, err := ts.GetTablet(ctx, primary)
+		if err != nil {
+			return err
+		}
+
+		var selectQueries []string
+		for _, t := range tables {
+			selectQueries = append(selectQueries, fmt.Sprintf("(select '%s' from %s limit 1)", t, t))
+		}
+		query := strings.Join(selectQueries, "union all")
```

Comment:
Why do this for every shard? Can't we build the query once and then use it on each shard?

Keep in mind that the size of this query is unbounded, so we could hit `max_allowed_packet`. I want to say that there's also a limit on the number of UNIONs you can do in a single statement but I didn't find any docs on that in a quick search. For JOINs e.g.:
```
The maximum number of tables that can be referenced in a single join is 61. This includes a join handled by merging derived tables and views in the FROM clause into the outer query block (see [Section 10.2.2.4, “Optimizing Derived Tables, View References, and Common Table Expressions with Merging or Materialization”](https://dev.mysql.com/doc/refman/8.4/en/derived-table-optimization.html)).
```

So that may be what we bump up against here.

Reply:
Thanks Matt for finding this. Fixed that. But even I couldn't find out the limit of `UNION`s on any docs or web search. What do you think should be the limit? Should we break the query down into multiple queries?